scalafix-core/shared/src/main/scala/scalafix/patch/Patch.scala

```diff
@@ -198,6 +198,12 @@ object Patch {
     patch.isEmpty || hasLintMessage && onlyLint
   }
 
+  private[scalafix] def isAllAtomic(patch: Patch): Boolean = patch match {
+    case Concat(a, b) => isAllAtomic(a) && isAllAtomic(b)
+    case AtomicPatch(_) => true
+    case _ => false
+  }
+
```

Comment:
I just realized that `isAllAtomic` would return false for `EmptyPatch`, `LintPatch`, ... which is probably not correct. In fact what we want to detect is the presence of top-level `TokenPatch`es which would cause rules not to respect the suppression comments/annotation. I think we can change it a bit to:

```scala
  private[scalafix] def isAllTokenPatchAtomic(patch: Patch): Boolean = patch match {
    case Concat(a, b) => isAllTokenPatchAtomic(a) && isAllTokenPatchAtomic(b)
    case _: TokenPatch => false
    case _ => true
  }
```

@olafurpg , what do you think?

Reply:
Indeed reversing the cases should help